sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/RowDeltaUtils.scala

```diff
@@ -25,4 +25,5 @@ object RowDeltaUtils {
   final val DELETE_OPERATION: Int = 1
   final val UPDATE_OPERATION: Int = 2
   final val INSERT_OPERATION: Int = 3
+  final val ORIGINAL_ROW_ID_VALUE_PREFIX: String = "__original_row_id_"
```

Comment:
can we briefly introduce how this works?

Reply:
It is needed to support modifying row ID values. I wasn't sure whether to support this but then I tested Postgres and I was able to modify the primary key value. I have a few relevant comments in `RewriteUpdateTable` and [here](https://github.com/apache/spark/pull/41028/files#r1183777434). It is essentially a column prefix used to project the original row ID value if it is modified. Let me me know if that makes sense.